uint32 vehicles
VehicleState[] vehicle_states
bool at_junction
bool register
bool remove
bool interaction